src/js/angular/core/directives.js

```diff
@@ -257,20 +257,43 @@ function searchResourceInput($location, toastr, ClassInstanceDetailsService, Aut
             uriValidation: '@',
             preserveInput: '@',
             empty: '=',
-            openInNewTab: '@'
+            openInNewTab: '@',
+            preserveSearch: '@',
+            radioButtons: '@',
+            clearInputIcon: '@'
         },
         templateUrl: 'js/angular/core/templates/search-resource-input.html',
         link: function ($scope, element, attrs) {
             element.autoCompleteStatus = undefined;
             element.autoCompleteWarning = false;
             $scope.empty = false;
-            $scope.searchInput = "";
             const MIN_CHAR_LEN = 0;
-
             // use a global var to keep old uri in order to change it when a new one appears
             let expandedUri;
-
             let canceler;
+            $scope.showClearInputIcon = false;
+            $scope.searchType = localStorage.getItem('searchType') || "table";
```

Comment:
Use the `local-storage-adapter.js` please!

Reply:
And also you should scope them a bit, e.g. use keys like `rdfsearchinput.searchtype` or something in order to prevent collisions.